src/io/EngineLogAccess.hpp

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The XCSoar Project
+
+#pragma once
+
+#include "io/FileOutputStream.hxx"
+
+#include <map>
+#include <string>
+#include <string_view>
+
+/**
+ * Engine Runtime Log access. For reading and writing engine runtime to a file.
+ * The log file might have duplicated entries for the key,
+ * with different values. Key is the BT MAC of a engine sensor,
+ * with engine runtime in seconds, as value.
+ */
+class EngineLogAccess{
+private:
+/**
+ * Filename of the LocalPath file.
+ */
+constexpr std::string_view log_file_name{"engine_runtime.log"};
```

Comment:
This will fail to compile on Windows. Also this allocates 16 bytes in each instance for no reason.

Reply:
Should there be more than one instance of this?
What would be a better type?

Reply:
I don't understand the question - it was your choice to make this a per-instance variable, so it's you who should answer that question.
The canonical type for file name literals is `Path`.